flocker/control/test/test_model.py

```diff
@@ -1460,3 +1462,190 @@ def test_case_insensitive(self):
+    """
+    Tests for ``Leases``.
+    """
+    def setUp(self):
+        """
+        Setup for each test.
+        """
+        self.leases = Leases()
+        self.now = datetime.datetime.now()
+        self.dataset_id = uuid4()
+        self.node_id = uuid4()
+        self.dataset = Dataset(dataset_id=unicode(self.dataset_id))
+        self.node = Node(uuid=self.node_id)
+        self.lease_duration = 60 * 60
+
+    def test_lease_expiry_datetime(self):
+        """
+        An lease has an expiry date/time after the specified number
+        of seconds from the time of acquisition.
+        """
+        expected_expiration = self.now + datetime.timedelta(
+            seconds=self.lease_duration)
+        leases = self.leases.acquire(
+            self.now, self.dataset_id, self.node_id, self.lease_duration
+        )
+        lease = leases.get(self.dataset_id)
+        self.assertEqual(lease.expiration, expected_expiration)
+
+    def test_indefinite_lease(self):
+        """
+        An acquired lease can be set to never expire.
+        """
+        leases = self.leases.acquire(self.now, self.dataset_id, self.node_id)
+        lease = leases.get(self.dataset_id)
+        self.assertIsNone(lease.expiration)
+
+    def test_lease_expires(self):
+        """
+        An acquired lease expires after the specified number of seconds and
+        is removed from the ``Leases`` map.
+        """
+        leases = self.leases.acquire(
+            self.now, self.dataset_id, self.node_id, self.lease_duration
+        )
+        # Assert the lease has been acquired successfully.
+        self.assertTrue(self.dataset_id in leases)
+        # Fake a time the first lease has expired.
+        now = self.now + datetime.timedelta(seconds=self.lease_duration + 1)
+        leases = leases.expire(now)
+        # Assert the lease has been removed successfully.
+        self.assertFalse(self.dataset_id in leases)
+
+    def test_indefinite_lease_never_expires(self):
+        """
+        An acquired lease set to never expire is not removed from ``Leases``
+        map.
+        """
+        leases = self.leases.acquire(self.now, self.dataset_id, self.node_id)
+        self.assertTrue(self.dataset_id in leases)
+        now = self.now + datetime.timedelta(seconds=self.lease_duration)
+        leases = leases.expire(now)
+        self.assertTrue(self.dataset_id in leases)
+
+    def test_error_on_check_lease_held_by_other_node(self):
+        """
+        A ``ValueError`` is raised when checking a lease held by another node.
+        """
+        # Acquire a lease on a node
+        leases = self.leases.acquire(self.now, self.dataset_id, self.node_id)
+        # Check the lease for a different node UUID
+        exception = self.assertRaises(
+            ValueError, leases._check_lease, self.dataset_id, uuid4()
+        )
+        self.assertEqual(
+            exception.message, "Lease already held by another node"
+        )
```

Comment:
:question: Reasonable people disagree on this point, but I generally don't see the point of writing tests for internal methods, especially ones that are this simple. Consider deleting this test.